archivebox/parsers/generic_json.py

```diff
@@ -59,11 +66,15 @@ def parse_generic_json_export(json_file: IO[str], **_kwargs) -> Iterable[Link]:
             elif link.get('name'):
                 title = link['name'].strip()
 
+            tags = ''
+            if link.get('tags'):
+                tags = link.get('tags').replace(' ',',')
```

Comment:
we do support tags containing spaces (but not newlines), so it might be better to replace '\n' with ','

Reply:
I was working off the example line and what Pinboard's JSON export looks like, which has space-separated tags. A more ideal JSON file would probably have `"tags": [ "tag1", "tag2", "tag with space" ]`, so maybe this should check to see `link.get('tags')` gives us a list, but then we have to put it together with commas because `Link()` can't handle `tags` as a list yet. I figure this could be revisited when that's tackled from the Roadmap.

Cleaning the `\n` out of tags should probably happen at a lower level, otherwise it will just be duplicated in every parser or other place tags enter the system.

More test data based on real-world examples would probably be good.

Reply:
Yeah that sounds good, lets try to check if it's a comma-separated string, space-separated string, or array. I have seen all 3 in the wild.

And I agree we should definitely collect more parser test inputs and add them to `tests/`.

Reply:
Okay, new commit coming. Handles lists, assumes a single string with any commas is already comma-separated, and anything else is space-separated so we turn those into commas. The edge case we'll mishandle is a JSON file that assumes the single-string version of `tag` is comma-separated and an entry only has one tag with a space in it. It probably makes the most sense to just get that wrong until someone provides evidence a tool that actually produces JSON files like that.

Maybe we could even be more clever by using `TAG_SEPARATOR_PATTERN` but that starts to feel like going overboard and I have a feeling that is a whole can of worms that nobody should open.

Reply:
Yup! My thoughts exactly, that edge case is probably rare.

Thanks for doing all this btw! This is a great fix / improvement to a fairly stale and brittle area of the codebase.